models/lstm.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+from tensorflow.keras.preprocessing import timeseries_dataset_from_array
+from tensorflow.keras.models import Sequential
+from tensorflow.keras.layers import LSTM, Dense
+import matplotlib.pyplot as plt
+
+def prepare_lstm_data(df, stock_1, stock_2, input_sequence_length=24, batch_size=32, split_fraction=0.8):
+    """
+    Prepara i dati per il modello LSTM.
+    Divide i dati in train e test set utilizzando una funzione di Keras.
+    """
+    # Estrai i valori per i due titoli selezionati
+    data_stock_1 = df[stock_1].values.reshape(-1, 1)
+    data_stock_2 = df[stock_2].values.reshape(-1, 1)
+
+    # Combina i dati in un unico array (stock_1 e stock_2 come feature)
+    data = np.hstack([data_stock_1, data_stock_2])
+
+    # Dividi il dataset in training e test
+    num_samples = len(data)
+    train_size = int(split_fraction * num_samples)
+
+    train_dataset = timeseries_dataset_from_array(
+        data=data[:train_size],
+        targets=data[input_sequence_length:train_size + input_sequence_length, 0],
```

Comment:
Are you sure input_sequence_length:train_size + input_sequence_length stays in the bounds of the data DataFrame?